File: Strivers/src/basicsmaths/primeornot/PrimeOrNot.java
package basicsmaths.primeornot;

import java.util.Scanner;

public class PrimeOrNot {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int num = sc.nextInt();
        System.out.println(primeOrNot(num));
    }

    private static boolean primeOrNot(int number) {
        for (int i = 2; i <= Math.sqrt(number); i++) {
            if (number % i == 0) return false;
        }
        return true;
    }
}

File: Strivers/src/basicsmaths/primeornot/PrimeOrNot.kt
package basicsmaths.primeornot

import java.util.Scanner

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val num = sc.nextInt()
    println(primeOrNot(num))
}

fun primeOrNot(number: Int): Boolean {
    for (i in 2..Math.sqrt(number.toDouble()).toInt()) {
        if (number % i == 0) return false
    }
    return true
}